mne/preprocessing/ica.py

```diff
@@ -120,7 +120,23 @@ class ICA(ContainsMixin):
     u"""M/EEG signal decomposition using Independent Component Analysis (ICA).
 
     This object can be used to estimate ICA components and then remove some
-    from Raw or Epochs for data exploration or artifact correction.
+    from Raw, Epochs or Evoked for data exploration or artifact correction.
+    
+    Prior to fitting and applying the ICA, data is whitened (de-correlated and
+    scaled to unit variance, also called sphering transformation) by means of
+    a Principle Component Analysis (PCA). In addition to the whitening, this step
+    introduces the option to reduce the dimensionality of the data, both, prior to
+    fitting the ICA and prior to reverting to sensor space, controllable by the
+    two parameters `max_pca_components` and `n_pca_components`, respectively.
+    .. note:: Commonly used for reasons of i) computational efficiency and
+              ii) additional noise reduction, it is a matter of current debate
+              whether pre-ICA dimensionality reduction could decrease the reliability
+              and stability of the ICA, at least for EEG data and especially during
```

Comment:
IIRC the `[5]` paper does not show reduction in ICA reliability if one chooses to keep PCA components explaining 95% of the variance, only going below 95% is detrimental, right? If so - you could add a note about this here.

Reply:
No, they are saying you shouldn't do it at all: "PCA rank reduction (even if removing only 1% of data variance) adversely affected the dipolarity and stability of ICs accounting for potentials arising from brain and known non-brain processes."

But it is true that they only used 72 channel data, although I'm pretty sure their finding generalizes to more channels. The problem is that physiological sources might be very low variance, and thus PCA reduction might get rid of these sources.

Reply:
@cbrnr Ok, I take that 95% back, but 99% doesn't look so bad in the figures. And it depends on your use case - if you care only about removal of artifactual activity with ICA (this is what I almost always do) then the fact that you get less dipolar brain sources is not so relevant (at least to the extent that it does not influence artifact cleaning).

Reply:
@agramfort they are not saying PCA is bad, and of course they use it to initialize. Their argument is about dropping PCA components, which in their opinion you should not do.

Reply:
yes but when you use EEGLAB if you don't remove PCA components then you don't run PCA so you use a different initialization of the unmixing matrix. My hunch is that this has a huge impact on the claims of this paper.

Reply:
I'm not sure this is true. I think it does run PCA by default to initialize. But I agree if they didn't this would make a huge difference.

Reply:
What's the difference between sphering and PCA?

Reply:
see paragraph "Relationship between initialization and dipolarity." in https://hal.archives-ouvertes.fr/hal-01451432/document as well as figure 1.

Brutal copy paste from pdf:
Wpca = D−1/2U, Wsph = UD−1/2U.

Reply:
Thanks, I always mix up sphering and whitening. (Say it with me: Both sphering and PCA are whitening transforms.) Very nice paper too - I think I'll take a closer look at Picard for my next EEG analysis.

Are you planning to address this flaw with a rebuttal? It would be nice to read some kind of published document, because EEGLAB-related stuff by these authors tend to be very visible in the community.

Reply:
I know but my time is too limited to start this. First we would need to replicate the study and show that using PCA full rank (and not sphering) breaks the conclusions. Too many stuff on my plate already.

Reply:
I can relate 😄 - but you already have the preprint out there, which I really like. One quick question: LBFGS is improving the ICA results by pushing the convergence. This is what EEGLAB folks are doing with AMICA, where they combine gradient descent and quasi-Newton optimization. Is this available somewhere within MNE? Is this what Picard does?

Reply:
AMICA uses a quasi Newton with Hessian approximation (which is not perfect) and what Picard does it using L-BFGS to correct the rough Hessian approximation. Now AMICA does more. It estimates the density distribution of the sources while Picard uses the same fixed non-linearities/densities than Infomax. Also AMICA does some unclear magic to omit outlier samples during the estimation. It is still unclear if AMICA is considered better than Infomax because 1) it estimates the densities 2) uses quasi Newton or 3) is robust to artifacts.

Reply:
This PR is a bit weird place to continue this discussion (if you want we could continue over email), but I thought a bit more about your comments @agramfort. Since the terms kept confusing me, I found that what you are referring to as sphering is more commonly called ZCA. Thus, de-correlating data is called [whitening or sphering](https://en.wikipedia.org/wiki/Whitening_transformation) (these terms can be used interchangeably). One specific method to whiten is PCA. Another method is ZCA. These two methods are related in a sense that ZCA is PCA followed by a rotation back. [This SO answer](https://stats.stackexchange.com/a/117463/53514) summarizes this nicely.

I will think more on the implications of these two methods for the initialization of ICA. Let me know if and how you would like to continue this discussion.